Fix RTL issues with discussion sass

andy-armstrong opened the pull request:
### Description
 
[TNL-4709](https://openedx.atlassian.net/browse/TNL-4709)

This change updates all left/right CSS rules in discussions to use bi-app functions so that they correctly handle switching to right-to-left.

### Sandbox
- [ ] https://andy-armstrong.sandbox.edx.org/courses/course-v1:edX+DemoX+Demo_Course/discussion/forum/

### Testing
- [ ] RTL

### Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [x] Code review: @bjacobel 
- [x] Code review: @alisan617
- [ ] Accessibility review: @clrux

FYI: @OmarIthawi 

### Post-review
- [ ] Rebase and squash commits

alisan617 commented:
@andy-armstrong the header has some extra links and user drop down menu arrow is off.
<img width="1229" alt="screen shot 2016-09-08 at 9 24 22 am" src="https://cloud.githubusercontent.com/assets/16127998/18350971/5219c2fa-75a6-11e6-984e-57167393a826.png">

bjacobel commented:
@alisan617 Those issues were preexisting in master - the dropdown one is already a bug [TNL-5073](https://openedx.atlassian.net/browse/TNL-5073) but the other one probably needs to be filed as a bug with [this feature](https://groups.google.com/a/edx.org/forum/#!topic/141portland/CtK4W5XjWPc) developed by the docs team (IDK what project that goes into).

bjacobel commented on lms/static/sass/discussion/_discussion.scss:
```diff
@@ -19,7 +19,7 @@ body.discussion {
     .post-cancel {
       @include float(left);
-      margin: ($baseline/2) 0 0 ($baseline*0.75);
+      @include margin(($baseline/2), 0, 0, ($baseline*0.75));
```
nit: our sass styleguide says to use spaces around operators, eg `($baseline / 2)`

bjacobel commented on lms/static/sass/discussion/utilities/_shame.scss:
```diff
@@ -129,7 +129,6 @@ li[class*=forum-nav-thread-label-] {
   // Override courseware
   .post-actions-list, .response-actions-list, .comment-actions-list {
     @extend %t-copy-sub2;
-    padding-left: 0 !important;
```
:thumbsup: guess this wasn't !important after all...

bjacobel commented:
Thanks for tackling this one @andy-armstrong, I'm super guilty of adding these left/right things in without thinking about it.

There will still be one issue with the forums after this is merged with the search box, I filed that in [FEDX-228](https://openedx.atlassian.net/browse/FEDX-228) a while back. Fixing it will require a UXPL update which I know is a loaded subject right now...

alisan617 commented:
@bjacobel thanks for clarifying.

I notice the filter icon is absolute positioned to the right in RTL.

in _navigation.scss line 51. Btw. the calculation is super.
```
.forum-nav-browse-filter .icon {
     right: ($baseline/4 + 1px + $baseline / 4); // Wrapper padding + border + input padding
}
```

andy-armstrong commented:
@alisan617 thanks for noticing the left/right rules. I've addressed all of those now, and I also went through all the rules listed in the bi-app documentation (http://anasnakawa.github.io/bi-app-sass/) and fixed a few more too. I've also rebased and am updating my sandbox, so hopefully that will fix the two issues with the header that you saw.

alisan617 commented:
Not sure if the wmd toolbar styles is in this PR's scope. The 3 | are not aligned correctly. For example:
`body.discussion .wmd-spacer3 {
    right: 300px;
}`
Nitpick on the icon alignment -- that looks like coming from the .btn-link .icon { margin-left } if this comes from UXPL, please ignore this comment.

<img width="759" alt="screen shot 2016-09-09 at 12 22 50 pm" src="https://cloud.githubusercontent.com/assets/16127998/18394641/de0db888-7688-11e6-9434-69015d995369.png">

bjacobel commented:
Sorry, just realized I forgot to give an explicit :thumbsup: in my comment above - this looks good from my end

andy-armstrong commented:
@alisan617 thanks for the review. I've fixed the separators now since making the toolbar RTL-compatible is out-of-scope. As we found out together, the icon alignment is a bug in the pattern library's RTL support, so I'll fix that separately.

I've updated my sandbox, so please re-review. Thanks.

alisan617 commented:
LGTM 👍
